ruff_lsp/server.py

```diff
@@ -1209,15 +1216,21 @@ async def apply_format(arguments: tuple[TextDocument]):
 
 
 @LSP_SERVER.feature(TEXT_DOCUMENT_FORMATTING)
-async def format_document(params: DocumentFormattingParams) -> list[TextEdit] | None:
+async def format_document(
+    params: DocumentFormattingParams, range: Range | None = None
+) -> list[TextEdit] | None:
     # For a Jupyter Notebook, this request can only format a single cell as the
     # request itself can only act on a text document. A cell in a Notebook is
     # represented as a text document. The "Notebook: Format notebook" action calls
     # this request for every cell.
     document = Document.from_cell_or_text_uri(params.text_document.uri)
     settings = _get_settings_by_document(document.path)
 
-    result = await _run_format_on_document(document, settings)
+    # We don't support range formatting of notebooks yet but VS Code
+    # doesn't seem to respect the document filter. For now, format the entire cell.
+    range = None if document.kind is DocumentKind.Cell else range
```

Comment:
This is unfortunate because as a user I wouldn't expect that "Format selection" in Jupyter notebook would format an entire cell. I'm not sure what the right solution is but I'd prefer to send a info / warning message to the user stating that range formatting isn't yet supported for Jupyter Notebooks instead.

Reply:
I agree but I wasn't able to come up with something better. I'm worried that a warning would be rather annoying and force users to change their setting from `modificationsIfAvailable` to the entire document or they get annoyed by the warning on every save